packages/wifi-unstuck-wa/files/usr/lib/lua/wifi_unstuck_wa.lua

```diff
@@ -28,8 +31,48 @@ function wu.get_stickable_ifaces()
 	return ifaces
 end
 
+function wu.wait_and_kill_on_timeout(pid_time_started)
+	local pid_done = {}
+
+	for pid,time_started in pairs(pid_time_started) do
+		pid_done[pid]=false
+	end
+
+	repeat
+		-- wait for 100ms
+		nixio.nanosleep(0,100e6)
+
+		-- see if something changed
+		while true do
+			pid,state,code = nixio.waitpid(nil,"nohang")
+			if not pid then break end
```

Comment:
From the documentation on https://neopallium.github.io/nixio/modules/nixio.html#nixio.waitpid I don't understand when `pid` is going to be `nil`. There it only says that pid can be zero, but in Lua zero in an if sentence is a true, so I don't understand how this works. But it works!

Reply:
That's because it returns `nil,nil,nil` if no child has changed state, despite documentation says otherwise. Anyway, one shouldn't rely on this. It will improve it.